apps/data_warehouse/lib/data_warehouse.ex

```diff
@@ -4,23 +4,24 @@ defmodule DataWarehouse do
   """
   alias DataWarehouse.Subscriber.DynamicSupervisor
 
-  def start_storing(stream, symbol) do
-    to_topic(stream, symbol)
+  def start_storing(stream, symbol, interval \\ "") do
+    to_topic(stream, symbol, interval)
     |> DynamicSupervisor.start_worker()
   end
 
-  def stop_storing(stream, symbol) do
-    to_topic(stream, symbol)
+  def stop_storing(stream, symbol, interval \\ "") do
+    to_topic(stream, symbol, interval)
     |> DynamicSupervisor.stop_worker()
   end
 
   def publish_data(args) do
     DataWarehouse.Publisher.start_link(args)
   end
 
-  defp to_topic(stream, symbol) do
+  defp to_topic(stream, symbol, interval) do
     [stream, symbol]
     |> Enum.map(&String.upcase/1)
     |> Enum.join(":")
+    |> Kernel.<>(interval)
```

Comment:
This is kind of hard to figure out what the final string looks like. Maybe just use string interpolation?

```suggestion
  defp to_topic(stream, symbol, interval) do
    "#{String.upcase(stream)}:#{String.upcase(symbol}#{interval}"
```

Reply:
simplicity is the ultimate sophistication

Reply:
Normally I would -1 this, since this is maybe due to unfamiliarity with elixir. We join strings using `<>` operator. Kernel is just the imported module for language level stuff. 

Fun fact. All operators in functional languages, Elixir included, are just functions. 

But I agree, in this case in particular string interpolation is just better

Reply:
I mean, it would be ok-ish to do a `"a" ++ "b"` in most cases, instead of ``${a}${b}`` (javascript). Case stands either way